src/agentscope/browser/web_browser.py

```diff
@@ -0,0 +1,366 @@
+        )
+
+        labels_handle = result_handle.get_property("labels")
+        items_handle = result_handle.get_property("items")
+
+        items_js_handles = (
+            items_handle.evaluate_handle("items => items")
+            .get_properties()
+            .values()
+        )
+        elements = [
+            item.get_property("element").as_element()
+            for item in items_js_handles
+            # if item.get_property("element").as_element()
+        ]
+
+        self.page_elements = elements
+
+        items_raw = items_handle.json_value()
+        web_ele_infos = [
+            self.page.evaluate("getElementInfo", ele) for ele in elements
+        ]
+
+        format_ele_text = []
+        input_attr_types = ["text", "search", "password", "email", "tel"]
+        for web_ele_id in range(len(elements)):
+            label_text = items_raw[web_ele_id]["text"]
+            ele_tag_name = web_ele_infos[web_ele_id]["tag_name"]
+            ele_type = web_ele_infos[web_ele_id]["type"]
+            ele_aria_label = web_ele_infos[web_ele_id]["aria_label"]
+            ele_meta_data = web_ele_infos[web_ele_id]["meta_data"]
+            meta_string = " ".join(ele_meta_data)
+            meta = f" {meta_string}"
+
+            # TODO for page too long, we should add selection
+            # on whether the element is in view port
+            if with_select and not (
+                (
+                    not label_text
+                    and (
+                        (
+                            ele_tag_name == "input"
+                            and ele_type in input_attr_types
+                        )
+                        or ele_tag_name == "textarea"
+                        or (
+                            ele_tag_name == "button"
+                            and ele_type in ["submit", "button"]
+                        )
+                    )
+                )
+                or (
+                    label_text
+                    and len(label_text) < 200
+                    and not ("<img" in label_text and "src=" in label_text)
+                )
+            ):
+                continue
+
+            message = f"[{web_ele_id}]: <{ele_tag_name}>"
+            if meta and with_meta:
+                message += f"<meta data: {meta}>"
+            if ele_aria_label:
+                message += f'"<{ele_aria_label}>"'
+            if label_text:
+                message += f'"text: {label_text}"'
+            format_ele_text.append(message)
+
+        screenshot_bytes = None
+
+        if vision:
+            screenshot_bytes = self.page_screenshot()
+            self._remove_labels_by_handle(labels_handle)
+
+        time.sleep(5)
+
+        return elements, format_ele_text, screenshot_bytes, web_ele_infos
+
+    @property
+    def url(self) -> str:
+        """return the url of current page"""
+        return self.page.url
+
+    @property
+    def page_html(self) -> str:
```

Comment:
@ZiTao-Li I remember that some certain service functions in AgentScope have already achieved similar functionalities?
We should consider how to reuse it or unified them rather than achieving the same functionalities in different places.

Reply:
I believe it's the `load_web` function you are talking about, which use `requests` package to get certain content from an url. 
Given that the playwright browser module already have built in method to get the html of a page, I think it is unnecessary to use `requests` module to get the html content of a page again, it will only cause unnecessary overheads and confusion.
However, we might include the `digest_webpage` if it yields good performance.